### PR TITLE
Improvements in scale tween + on complete event for sequences

### PR DESCRIPTION
**Pull request by luismoyano**
It adds OnComplete event to the tween sequences so buttons can perform their actions after the jumpy animation is done, also adds float3 support for LOScale Tween

Preview

Check with gameplay branch feature/menu_anims

### CODE REVIEW

File: Engine/Animation/Tween/Tween.cpp [hunk 1]
```diff
@@ -223,42 +223,51 @@ void Tween::ResetTween()
...
-float Tween::EaseInSine(float t) const {
+float Tween::EaseInSine(float t) const 
+{
 	return 1 - cos(t * (3.1416f / 2));
 }
 
-float Tween::Sine(float t) const {
+float Tween::Sine(float t) const 
+{
 	return (sin(t * (3.1416f * 2) - 3.1416f / 2) + 1.0f) / 2.0f;
```
Comment (mariofv) [inline Engine/Animation/Tween/Tween.cpp]: 3.1416f should be define as constant called PI

File: Engine/Animation/Tween/Tween.h [hunk 2]
```diff
@@ -52,7 +52,7 @@ class Tween
 
 	ENGINE_API static Tween* LOTranslate(ComponentTransform2D*, float2, float);
 	ENGINE_API static Tween* LORotate(ComponentTransform2D*, float, float);
-	ENGINE_API static Tween* LOScale(ComponentTransform2D*, float, float);
+	ENGINE_API static Tween* LOScale(ComponentTransform2D*, float3, float);
```
Comment (mariofv) [inline Engine/Animation/Tween/Tween.h]: Varaible names are missing inside declarations

**Review by mariofv — COMMENTED** [review]
@luismoyano I'm trying this branch with [this](https://github.com/OnionGalaxy/TwinlightSouls/pull/52) branch, but I cannot execute the engine. Which should I use?

File: Engine/Animation/Tween/Tween.cpp [hunk 3]
```diff
@@ -40,12 +40,12 @@ Tween * Tween::LORotate(ComponentTransform2D* transform, float end_value, float
 	return tween;
 }
 
-Tween * Tween::LOScale(ComponentTransform2D* transform, float end_scale, float desired_time)
+Tween * Tween::LOScale(ComponentTransform2D* transform, float3 end_scale, float desired_time)
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp]: end_scale can be const ref right?

File: Engine/Animation/Tween/Tween.cpp [hunk 4]
```diff
@@ -223,42 +223,51 @@ void Tween::ResetTween()
 	}
 }
 
-float Tween::Linear(float t) const {
+float Tween::Linear(float t) const
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp:226]: const float t maybe?

File: Engine/Animation/Tween/TweenSequence.cpp [hunk 5]
```diff
@@ -102,6 +108,7 @@ void TweenSequence::Update(float dt)
 
 	if (current_played_tweens.size() <= 0)
 	{
+		if (on_completed_callback != nullptr) on_completed_callback();
```
Comment (Dartemiss) [inline Engine/Animation/Tween/TweenSequence.cpp]: CS, 
if(blabal) 
{
    xxxxxx
}

**Review by Dartemiss — APPROVED** [review]
All good my friend! GJ

File: Engine/Animation/Tween/Tween.cpp [hunk 6]
```diff
@@ -16,7 +16,7 @@ void Tween::Pause()
 	state = TweenState::PAUSED;
 }
 
-Tween * Tween::LOTranslate(ComponentTransform2D* transform, float2 end_value, float desired_time)
+Tween * Tween::LOTranslate(ComponentTransform2D* transform, const float2 end_value, float desired_time)
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp:19]: Code style: Tween*

File: Engine/Animation/Tween/Tween.cpp [hunk 7]
```diff
@@ -28,7 +28,7 @@ Tween * Tween::LOTranslate(ComponentTransform2D* transform, float2 end_value, fl
 	return tween;
 }
 
-Tween * Tween::LORotate(ComponentTransform2D* transform, float end_value, float desired_time)
+Tween * Tween::LORotate(ComponentTransform2D* transform, const float end_value, float desired_time)
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp:31]: Code Style

File: Engine/Animation/Tween/Tween.cpp [hunk 8]
```diff
@@ -40,19 +40,19 @@ Tween * Tween::LORotate(ComponentTransform2D* transform, float end_value, float
 	return tween;
 }
 
-Tween * Tween::LOScale(ComponentTransform2D* transform, float end_scale, float desired_time)
+Tween * Tween::LOScale(ComponentTransform2D* transform, const float3 end_scale, float desired_time)
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp:43]: Maybe can be a const reference? (end_scale)

File: Engine/Animation/Tween/TweenSequence.cpp [hunk 9]
```diff
@@ -96,12 +102,17 @@ void TweenSequence::Update(float dt)
 			current_played_tweens.erase(current_played_tweens.begin() + i);
 			i -= 1;
 
-			delete(the_tween);
+			//delete(the_tween);
```
Comment (Dartemiss) [inline Engine/Animation/Tween/TweenSequence.cpp]: Why is this commented?

**Review by Dartemiss — APPROVED** [review]
Good job man! Very clever. I even learnt how to use std::function (never used it before).
Just a few comments above.

File: Engine/Animation/Tween/TweenSequence.cpp (hunk 9 above)
Comment (luismoyano) [reply Engine/Animation/Tween/TweenSequence.cpp]: Uncommented, it was my mistake

**Review by vidalmelero — APPROVED** [review]

**Pull request merged**